### DISCUSSION TITLE
Compatible with ESP32?

### DISCUSSION THREAD

**Discussion by vintageradio** [FortySevenEffects/arduino_midi_library#242] [Q&A] [answered]
For various reasons I need to host my MIDI controller on an ESP32.  Is this library compatible? Has anyone gotten it working on an ESP32?  Thanks!

**Comment by lathoub** [comment, marked as answer]
ESP32 has 3 UARTs - I'm using Serial1 and Serial2 without an issue. So: yes!
(I do not recommend using Serial, as it is already used)

Usage:
`MIDI_CREATE_INSTANCE(HardwareSerial, Serial1, MIDI);`

**Reply by vintageradio** [reply]
Great! - I'll give that a try. Thanks!
